src/node/cliConfig.ts

```diff
@@ -0,0 +1,58 @@
+import { asObject, asOptional, asString } from 'cleaners'
+import fs from 'fs'
+import { join, resolve } from 'path'
+import xdgBasedir from 'xdg-basedir'
+
+export interface CliConfig {
+  apiKey?: string
+  appId?: string
+  authServer?: string
+  directory?: string
+  password?: string
+  username?: string
+  workingDir?: string
+}
+
+const asCliConfig = asObject<CliConfig>({
+  apiKey: asOptional(asString),
+  appId: asOptional(asString),
+  authServer: asOptional(asString),
+  directory: asOptional(asString),
+  password: asOptional(asString),
+  username: asOptional(asString),
+  workingDir: asOptional(asString)
+})
+
+export function loadConfig(configPath?: string): CliConfig {
+  let where: string | undefined
+  let text: string | undefined
+
+  // Try to load the file:
+  if (configPath != null) {
+    try {
+      where = resolve(configPath)
```

Comment:
I guess resolve doesn't need __dirname; it defaults to local dirname

Reply:
Yep: https://nodejs.org/api/path.html#pathresolvepaths

Reply:
Yep: https://nodejs.org/api/path.html#pathresolvepaths